examples/04_scalability.py

```diff
@@ -197,3 +200,55 @@ def wrapped_func(*args, **kwargs):
 [t.set(size=16) for t in ax2.get_yticklabels()]
 plt.tight_layout()
 
+##################################################################################
+# SimilarityEncoder with default options and 32 bits float dtype
+# ---------------------------------------------------------------
+#
+# As done above we build our vectorizer using the ColumnTransformer
+# to combine one-hot encoding and similarity encoding.
+# We set the dtype parameters for both encoders to np.float32
+# and cast the column 'year' to np.float32
+# (to prevent the ColumnTransformer from force casting everything to float64).
+
+sim_enc = SimilarityEncoder(similarity='ngram', dtype=np.float32, handle_unknown='ignore')
+
+y = df['Violation Type']
+df['Year'] = df['Year'].astype(np.float32)
+
+transformers = [('one_hot', OneHotEncoder(sparse=False, dtype=np.float32, handle_unknown='ignore'),
+                 ['Alcohol',
+                  'Arrest Type',
+                  'Belts',
+                  'Commercial License',
+                  'Commercial Vehicle',
+                  'Fatal',
+                  'Gender',
+                  'HAZMAT',
+                  'Property Damage',
+                  'Race',
+                  'Work Zone']),
+                ('pass', 'passthrough', ['Year']),
+                ]
+
+column_trans = ColumnTransformer(
+    # adding the dirty column
+    transformers=transformers + [('sim_enc', sim_enc, ['Description'])],
+    remainder='drop')
+
+t0 = time()
+X = column_trans.fit_transform(df)
+t1 = time()
+print('Type of vectorized data:', X.dtype)
+print('Time to vectorize: %s' % (t1 - t0))
+
+################################################################################
+# We can run a cross-validation
+model = pipeline.make_pipeline(column_trans, log_ref)
+results = resource_used(model_selection.cross_validate)(model, df, y)
+print("Cross-validation score: %s" % results['test_score'])
+
+################################################################################
+# Here we see that the memory consumption is reduced compared to the
+# case where 64 bits float are used.
+# Which is not the case with all the sci-kit learn models, where some
```

Comment:
It's spelled "scikit-learn" :).

As you can see from the example:
https://385-124964482-gh.circle-artifacts.com/0/doc/auto_examples/04_scalability.html
there is little gain in memory used during the cross-validation. This is because the LBFG solver does not work with float32 and casts in float64.

Can you try changing the whole example to use the "saga" solver. I think that it should be faster and work with 32 bit.

Reply:
I did and the score go way down (they are all around 0.7 against 0.78, 0.79 with LBFG). Should I stay with saga?